src/backend/distributed/utils/statistics_collection.c

```diff
@@ -0,0 +1,213 @@
+/*-------------------------------------------------------------------------
+ *
+ * statistics_collection.c
+ *	  Anonymous reports and statistics collection.
+ *
+ * Copyright (c) 2017, Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#if HAVE_LIBCURL
+
+#include "postgres.h"
+
+#include <curl/curl.h>
+#include <sys/utsname.h>
+
+#include "access/xact.h"
+#include "citus_version.h"
+#include "distributed/metadata_cache.h"
+#include "distributed/statistics_collection.h"
+#include "distributed/worker_manager.h"
+#include "lib/stringinfo.h"
+#include "utils/json.h"
+
+bool EnableStatisticsCollection = true; /* send basic usage statistics to Citus */
+
+static uint64 NextPow2(uint64 n);
+static uint64 ClusterSize(List *distributedTableList);
+static bool SendHttpPostJsonRequest(const char *url, const char *postFields, long
+									timeout_seconds);
+
+/* WarnIfSyncDNS warns if libcurl is compiled with synchronous DNS. */
+void
+WarnIfSyncDNS(void)
+{
+	curl_version_info_data *version_info = curl_version_info(CURLVERSION_NOW);
+	if (!(version_info->features & CURL_VERSION_ASYNCHDNS))
+	{
+		ereport(WARNING, (errmsg("your current libcurl version doesn't support "
+								 "asynchronous DNS, which might cause unexpected "
+								 "delays in database's operation"),
```

Comment:
Change to _delays in the operation of Citus_.

Reply:
Done.